hyperon_das_atomdb/adapters/redis_mongo_db.py

```diff
@@ -232,6 +235,30 @@ def _connection_redis(
 
         return self.redis
 
+    def _setup_indexes(self):
+        self.default_pattern_index_templates = []
+        for named_type in [True, False]:
+            for pos0 in [True, False]:
+                for pos1 in [True, False]:
+                    for pos2 in [True, False]:
+                        if named_type and pos0 and pos1 and pos2:
+                            # not a pattern but an actual atom
+                            continue
+                        template = {}
+                        template[MongoFieldNames.TYPE_NAME] = named_type
+                        template["selected_positions"] = []
```

Comment:
Suggestion for this part of the code:

`template["selected_positions"] = [i for i, pos in enumerate([pos0, pos1, pos2]) if pos]`